compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -734,6 +736,58 @@ pub struct CapturedPlace<'tcx> {
     pub info: CaptureInfo<'tcx>,
 }
 
+pub fn place_to_string_for_capture(tcx: TyCtxt<'tcx>, place: &HirPlace<'tcx>) -> String {
+    let name = match place.base {
+        HirPlaceBase::Upvar(upvar_id) => tcx.hir().name(upvar_id.var_path.hir_id).to_string(),
+        _ => bug!("Capture_information should only contain upvars"),
+    };
+    let mut curr_string = format!("{}", name);
+
+    for (i, proj) in place.projections.iter().enumerate() {
+        if HirProjectionKind::Deref == proj.kind {
```

Comment:
I'd just change this to a match block because we have four possible projections here: Deref, Field, Index, and Subslice. And then leave a bug for Index and Subslice saying that these projections aren't captured.

Now, this is a little concerning because once some of the bugs are fixed, we will see Index in the Place that is being used as the origin. 

Fix for this isn't too bad. I'll leave a FIXME comment in the correct place.

Reply:
Can you convert this into a match on proj.kind and not force Adts and bug out if the projection ins't applied to the correct type?